zio-http/shared/src/main/scala/zio/http/Response.scala

```diff
@@ -136,7 +136,7 @@ object Response {
   def error(status: Status.Error, message: String): Response = {
     import zio.http.internal.OutputEncoder
 
-    val message2 = OutputEncoder.encodeHtml(if (message == null) status.text else message)
+    val message2 = OutputEncoder.encodeHtml(if (message == null) status.text else message).replace("\n", " ")
```

Comment:
Might be helpful if this was replaced by `"\\n"` instead, I think it would make it easier to recover the original error text if that was ever desired. Also, I believe netty disallows other characters in its headers, would it make sense to handle all of them at once?

Reply:
Seems that escaping with '\\n' should lead to the same problem.

I think netty (or http spec) only accepts visible ascii characters (between 33 and 125),
should we use URL encoding to escape them ?